tools/protoc-plugin/src/main/java/io/spine/tools/protoc/method/MethodFactories.java

```diff
@@ -0,0 +1,165 @@
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.protoc.method;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.annotations.Immutable;
+import io.spine.logging.Logging;
+import io.spine.tools.protoc.Classpath;
+import io.spine.tools.protoc.MethodFactoryConfiguration;
+import io.spine.type.MessageType;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.util.Exceptions.newIllegalArgumentException;
+
+/**
+ * An utility class for instantiating {@link MethodFactory} for a particular
+ * {@link io.spine.tools.protoc.GeneratedMethod} specification.
+ */
+final class MethodFactories implements Logging {
+
+    private final ClassLoader externalClassLoader;
+
+    MethodFactories(MethodFactoryConfiguration configuration) {
+        this.externalClassLoader = externalClassLoader(configuration);
+    }
+
+    /**
+     * Creates an instance of a {@link MethodFactory} out of the supplied
+     * {@link io.spine.tools.protoc.GeneratedMethod specification}.
+     *
+     * <p>If specification is invalid or the specified class for some reason could not be
+     * instantiated a {@link NoOpMethodFactory} instance is returned.
+     */
+    MethodFactory newFactoryFor(String factoryName) {
+        checkNotNull(factoryName);
+        if (factoryName.trim()
+                       .isEmpty()) {
+            return NoOpMethodFactory.INSTANCE;
+        }
+        MethodFactory result = from(factoryName);
+        return result;
+    }
+
+    /**
+     * Instantiates a new {@link MethodFactory} from the specified fully-qualified class name.
+     */
+    private MethodFactory from(String fqn) {
+        Optional<Class<MethodFactory>> factoryClass = methodFactoryClass(fqn);
+        if (!factoryClass.isPresent()) {
+            return NoOpMethodFactory.INSTANCE;
+        }
+        try {
+            MethodFactory factory = factoryClass.get()
+                                                .getConstructor()
+                                                .newInstance();
+            return factory;
+        } catch (InstantiationException e) {
+            _warn("Unable to instantiate MethodFactory {}.", fqn, e);
```

Comment:
As far as I understand at the moment we don't fail the execution if something goes wrong. 

But to me such an issue is clearly a configuration mistake, and it should be fail-fast-detected. So I would throw an exception and stop the execution.